hnn_core/network_builder.py

```diff
@@ -425,35 +437,184 @@ def _create_cells_and_feeds(self):
-                                self.net.params)
-                cell.parreceive(gid, self.net.gid_dict,
-                                self.net.pos_dict, self.net.p_common)
-                # now do the unique external feeds specific to these cells
-                # parreceive_ext receives connections from UNIQUE
-                # external inputs
-                for cell_type in self.net.p_unique.keys():
-                    p_type = self.net.p_unique[cell_type]
-                    cell.parreceive_ext(
-                        cell_type, gid, self.net.gid_dict, self.net.pos_dict,
-                        p_type)
+        # source of synapse is always at soma
+
+        # layer2 Pyr -> layer2 Pyr
+        # layer5 Pyr -> layer5 Pyr
+        nc_dict['lamtha'] = 3.
+        for target_cell in ['L2Pyr', 'L5Pyr']:
+            for receptor in ['nmda', 'ampa']:
+                key = f'gbar_{target_cell}_{target_cell}_{receptor}'
+                nc_dict['A_weight'] = params[key]
+                self._connect_celltypes(target_cell, target_cell, 'proximal',
+                                        receptor, nc_dict,
+                                        allow_autapses=False)
+
+        # layer2 Basket -> layer2 Pyr
+        target_cell = 'L2Pyr'
+        nc_dict['lamtha'] = 50.
+        for receptor in ['gabaa', 'gabab']:
+            nc_dict['A_weight'] = params[f'gbar_L2Basket_L2Pyr_{receptor}']
+            self._connect_celltypes('L2Basket', target_cell, 'soma', receptor,
+                                    nc_dict)
+
+        # layer5 Basket -> layer5 Pyr
+        target_cell = 'L5Pyr'
+        nc_dict['lamtha'] = 70.
+        for receptor in ['gabaa', 'gabab']:
+            key = f'gbar_L5Basket_{target_cell}_{receptor}'
+            nc_dict['A_weight'] = params[key]
+            self._connect_celltypes('L5Basket', target_cell, 'soma', receptor,
+                                    nc_dict)
+
+        # layer2 Pyr -> layer5 Pyr
+        nc_dict['lamtha'] = 3.
+        for loc in ['proximal', 'distal']:
+            nc_dict['A_weight'] = params[f'gbar_L2Pyr_{target_cell}']
+            self._connect_celltypes('L2Pyr', target_cell, loc, 'ampa',
+                                    nc_dict)
+        # layer2 Basket -> layer5 Pyr
+        nc_dict['lamtha'] = 50.
+        nc_dict['A_weight'] = params[f'gbar_L2Basket_{target_cell}']
+        self._connect_celltypes('L2Basket', target_cell, 'distal', 'gabaa',
+                                nc_dict)
+
+        # xx -> layer2 Basket
+        target_cell = 'L2Basket'
+        nc_dict['lamtha'] = 3.
+        nc_dict['A_weight'] = params[f'gbar_L2Pyr_{target_cell}']
+        self._connect_celltypes('L2Pyr', target_cell, 'soma', 'ampa',
+                                nc_dict)
+        nc_dict['lamtha'] = 20.
+        nc_dict['A_weight'] = params[f'gbar_L2Basket_{target_cell}']
+        self._connect_celltypes('L2Basket', target_cell, 'soma', 'gabaa',
+                                nc_dict)
```

Comment:
Are autapses supposed to be allowed here but not for L5 basket cells below?

Reply:
it's what the old code says: https://github.com/jonescompneurolab/hnn-core/blob/master/hnn_core/basket.py#L78

and:

looks commented in HNN code too: 

https://github.com/jonescompneurolab/hnn/blob/master/L2_basket.py#L88

Reply:
I'm glad to see that cleaning up the code is unearthing these issues which will obviously lead to more consistent and better simulations

Reply:
Sure enough. @stephanie-r-jones do you know why L2 basket cells are allowed to synapse on themselves but the same isn't true for L5 basket cells?